packages/flutter/lib/src/material/tabs.dart

```diff
@@ -733,13 +729,14 @@ class _TabBarState<T> extends ScrollableState<TabBar<T>> implements TabBarSelect
 
     TextStyle textStyle = themeData.primaryTextTheme.body1;
     IconThemeData iconTheme = themeData.primaryIconTheme;
+    Color textColor = themeData.primaryTextTheme.body1.color.withAlpha(0xB2); // 70% alpha
 
     List<Widget> tabs = <Widget>[];
     bool textAndIcons = false;
     int tabIndex = 0;
     for (TabLabel label in config.labels.values) {
-      tabs.add(_toTab(label, tabIndex++, textStyle.color, indicatorColor));
-      if (label.text != null && label.icon != null)
+      tabs.add(_toTab(label, tabIndex++, textColor, indicatorColor));
+      if (label.text != null && (label.icon != null || label.builder != null))
```

Comment:
Yes it could return both however the builder is really only intended to be an alternative way to specify the icon - e.g. you could provide a CustomPaint.  I'll add some doc to explain as much and constrain the size of what the builder produces.